packages/api/src/lib/course-metadata-api.ts

```diff
@@ -0,0 +1,60 @@
+import axios from 'axios';
+import { getAuthHeaders } from './lmp';
+
+export interface LectureEntry {
+  lectureDay: string;
+  venue?: string;
+  venueLink?: string;
+  lectureStartTime: string;
+  lectureEndTime: string;
+  hasHomework?: boolean;
+  hasQuiz?: boolean;
+}
+
+export interface CourseMetadata {
+  courseId: string;
+  instanceId: string;
+  lectureSchedule: LectureEntry[];
+}
+
+const COURSE_METADATA_BASE_URL = '/api/course-metadata';
+
+export async function getLectureEntry(data: Pick<CourseMetadata, 'courseId' | 'instanceId'>) {
+  const response = await axios.get(`${COURSE_METADATA_BASE_URL}/get-lecture`, {
+    headers: getAuthHeaders(),
+    params: data,
+  });
+  return response.data;
+}
+
+export async function addLectureEntry(
+  data: Pick<CourseMetadata, 'courseId' | 'instanceId'> & { lectureEntry: LectureEntry }
```

Comment:
Create a new type and then use it in the api implementation too. Do this for other APIs too (where appropriate)


```suggestion
export type AddLectureEntryRequest = Pick<CourseMetadata, 'courseId' | 'instanceId'> & { lectureEntry: LectureEntry };
export async function addLectureEntry(
  data: UpdateHomeworkRequest,
```